python/src/opendp/extras/polars/__init__.py

```diff
@@ -996,40 +1073,124 @@ class Margin:
     Instances of this class are used by :py:func:`opendp.context.Context.compositor`.
     """
 
-    by: Sequence | None = None
+    by: Sequence = field(default_factory=list)
     """Polars expressions describing the grouping columns."""
 
-    public_info: Literal["keys"] | Literal["lengths"] | None = None
-    """Identifies properties of grouped data that are considered public information.
+    max_length: int | None = None
+    """An upper bound on the number of records in any one group.
+
+    If you don't know how many records are in the data, you can specify a very loose upper bound,
+    for example, the size of the total population you are sampling from.
```

Comment:
Should `max_length` be required? As a user, docstring makes me think so, but the typing puts that in doubt.